pandas/core/arrays/arrow/array.py

```diff
@@ -949,7 +951,68 @@ def _indexing_key_to_indices(
             indices = np.arange(n)[key]
         return indices
 
-    # TODO: redefine _rank using pc.rank with pyarrow 9.0
+    def _rank(
+        self: ArrowExtensionArrayT,
+        *,
+        axis: AxisInt = 0,
+        method: str = "average",
+        na_option: str = "keep",
+        ascending: bool = True,
+        pct: bool = False,
+    ) -> ArrowExtensionArrayT:
+        """
+        See Series.rank.__doc__.
+        """
+        if axis != 0:
+            raise NotImplementedError
+
+        if (
+            pa_version_under9p0
+            # as of version 10, pyarrow does not support an "average" method
+            or method not in ("min", "max", "first", "dense")
+        ):
+            from pandas.core.algorithms import rank
+
+            ranked = rank(
+                self.to_numpy(),
```

Comment:
Can't `ranked = super().rank(...)` because of the `to_numpy()` call? It seems that prior it works if `self` is passed?

Reply:
Yes, we could. However, I observed this:

```
import pandas as pd 
import pandas.core.algorithms as algos
import numpy as np

arr = pd.array(np.random.randn(10**6), dtype="float64[pyarrow]")

%timeit algos.rank(arr)
# 1.41 s ± 12.3 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

%timeit algos.rank(arr.to_numpy())
# 525 ms ± 9.83 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

Probably worth a look in `algos.rank` to see whats going on. Ok to do as a separate PR or should I look at that first?

Reply:
Follow up PR is good. Ideally it would be good to have the `axis != 1` case fall through to super and raise there